synthtool/gcp/templates/java_library/.kokoro/dependencies.sh

```diff
@@ -43,12 +43,13 @@ function completenessCheck() {
   # Output dep list with compile scope generated using the original pom
   # Running mvn dependency:list on Java versions that support modules will also include the module of the dependency.
   # This is stripped from the output as it is not present in the flattened pom.
+  # Only dependencies with 'compile' or 'runtime' scope are included from original dependency list.
   msg "Generating dependency list using original pom..."
-  mvn dependency:list -f pom.xml -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' | sed -e s/\\s--\\smodule.*// | grep -v ':test$' >.org-list.txt
+  mvn dependency:list -f pom.xml -DincludeScope=runtime -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' | sed -e s/\\s--\\smodule.*// >.org-list.txt
 
-  # Output dep list generated using the flattened pom (test scope deps are ommitted)
+  # Output dep list generated using the flattened pom (only 'compile' and 'runtime' scopes)
   msg "Generating dependency list using flattened pom..."
-  mvn dependency:list -f .flattened-pom.xml -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' >.new-list.txt
+  mvn dependency:list -f .flattened-pom.xml -DincludeScope=runtime -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' >.new-list.txt
```

Comment:
for the flatten one, should we not limit the scope? I.e., flattened pom should probably only have runtime artifacts even w/o scoping.

Reply:
compile time is needed too

Reply:
compile is needed too

Reply:
Based on the [doc](https://maven.apache.org/plugins/maven-dependency-plugin/list-mojo.html#includeScope):

runtime scope gives runtime and compile dependencies

Reply:
> for the flatten one, should we not limit the scope? I.e., flattened pom should probably only have runtime artifacts even w/o scoping.

This depends on what we do in the flatten plugin -- if we keep direct provides deps, flattened pom will have them. I think it is safer for us to do a strict comparison btw the original pom and the flattened pom here.

Reply:
This was my concern as well - I think making this change to the PR would depend on the status of the provided transitive PR on the flatten-maven-plugin, since we don't certainly know yet if direct provided dependencies will be included.